linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -2860,17 +2895,32 @@ private[emitter] class FunctionEmitter(sjsGen: SJSGen) {
           }
 
         case Closure(arrow, captureParams, params, body, captureValues) =>
-          val innerFunction = {
-            desugarToFunctionInternal(arrow, params, body, isStat = false,
-                Env.empty(AnyType).withParams(captureParams ++ params))
-          }
+          val capturesBuilder = List.newBuilder[(js.ParamDef, js.Tree)]
+          val replacementsBuilder = Map.newBuilder[LocalName, js.Tree]
 
-          val captures = for {
+          for {
             (param, value) <- captureParams.zip(captureValues)
           } yield {
-            (transformParamDef(param), transformExpr(value, param.ptpe))
+            val newValue = transformExpr(value, param.ptpe)
+
+            if (!param.mutable && mayElideIIFECapture(value, allowThis = arrow)) {
+              replacementsBuilder += param.name.name -> newValue
+            } else {
+              capturesBuilder += transformParamDef(param) -> newValue
+            }
           }
 
+          val innerFunction = {
+            val bodyEnv = Env.empty(AnyType)
+              .withInLoopForVarCapture(arrow && env.inLoopForVarCapture)
```

Comment:
Why `arrow &&`?

Reply:
Ah ah :) I was gonna write a comment and then realized it was wrong :)

My initial thought was that if it is not an arrow function, we'll introduce a new `var` scope (so far, the reasoning is correct). So therefore, all vars that we are seeing later are (from a scoping POV) not in a loop. This is correct from the IR POV, but not from a JS POV if an outer identifier evades capture :)

I removed it.